main.go

```diff
@@ -0,0 +1,45 @@
+package main
+
+import (
+	"bufio"
+	"flag"
+	"fmt"
+	"io"
+	"os"
+	"regexp"
+)
+
+func main() {
+	patternParameter := flag.String("e", "","use PATTERN for matching." )
+	flag.Parse()
+
+	if *patternParameter != "" {
+		re := regexp.MustCompile(*patternParameter)
+		buf := bufio.NewReader(os.Stdin)
+
+		for {
+			data, dataErr := buf.ReadBytes('\n')
+			if dataErr != nil && dataErr != io.EOF {
+				fmt.Fprintln(os.Stderr, dataErr)
+				os.Exit(1)
+			}
+
+			if len(data) == 0 {
+				break
+			}
+
+			if data[len(data) - 1] != '\n' {
+				data = append(data, '\n')
+			}
+
+			if re.Match(data) {
+				_, _ = os.Stdout.Write(data)
+			}
+
+			if dataErr == io.EOF {
+				break
+			}
+		}
+		return
+	}
```

Comment:
-e is a required parameter. How did you handle missing required parameters before?

Reply:
i print an error message and exit the program?

Reply:
Exactly.